### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

File: Utilities/types.h [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #define SAFE_BUFFERS
 #define NEVER_INLINE __attribute__((noinline))
 #define FORCE_INLINE __attribute__((always_inline)) inline
+#define thread_local __thread
```
Comment (danilaml) [inline Utilities/types.h]: Why for every non-msvc compiler and not just for MinGW? Also, afaik, it's not the same (rules for initialization differ from `thread_local`, though maybe it doesn't matter in our case).

Comment (Nekotekina) [inline Utilities/types.h]: Should be fine for all platforms.

Comment (danilaml) [inline Utilities/types.h]: @Nekotekina IMHO, better to use standard features if possible.

**Comment by Areiser** [comment]
Fixes both #2472 and #2462 (tested on arch linux)
Fps Counter

**Pull request merged**